src/hooks/useContract.ts

```diff
@@ -15,10 +16,12 @@ import { ERC20_BYTES32_ABI } from '../constants/abis/erc20'
 import PENDING_ABI from '../constants/abis/pending.json'
 import { getContract } from '../utils'
 import { useActiveWeb3React } from './useActiveWeb3React'
+import { useWeb3Context } from '@gooddollar/web3sdk-v2'
 
 // returns null on errors
 export function useContract(address: string | undefined, ABI: any, withSignerIfPossible = true): Contract | null {
-    const { library, account } = useActiveWeb3React()
+    const { account } = useActiveWeb3React()
+    const { web3Provider: library } = useWeb3Context() as { web3Provider: Web3Provider }
```

Comment:
just keep the library name... so you dont have to do all that name refactoring

Reply:
I went with web3provider because its more declarative for potential others who will utilize sdk-v2
don't you agree, else I either have to change it on the SDK side, or use a local hook for this context
@sirpy

Reply:
ok